php/Blockonomics.php

```diff
@@ -571,7 +581,8 @@ public function get_checkout_context($order, $crypto){
             } else {
                 // Display Checkout Page
                 $context['order_amount'] = $this->fix_displaying_small_values($order['satoshi']);
-                $context['payment_uri'] = $context['crypto']['uri'] . ":" . $order['address'] . "?amount=" . $context['order_amount'];
+                $context['payment_uri'] = $this->get_crypto_payment_uri($context['crypto'], $order['address'], $context['order_amount']);
+                $context['crypto_rate'] = $this->get_crypto_rate_from_params($order['value'], $context['order_amount']);
```

Comment:
Formatted numbers are strings so it better to change their variable name so that we are not confused 
['crypto_rate'] can be renamed to ['crypto_rate_str']
['order_amount'] can be renamed to ['order_btc_str']

Reply:
Also I think $this->get_crypto_rate_from_params should use $order['satoshi'] rather than $order['order_amount'] as we should calculate from the accurate satoshi value rather than the rounded string value